File: bluetooth_send.cpp
#include <iostream>
#include <bluetooth/bluetooth.h>
#include <bluetooth/rfcomm.h>
#include <unistd.h>
#include <cstring>

int main() {
    // Replace "00:11:22:33:44:55" with the MAC address of your Bluetooth device
    const char* dest = "00:11:22:33:44:55";

    // Create Bluetooth socket
    int s = socket(AF_BLUETOOTH, SOCK_STREAM, BTPROTO_RFCOMM);
    if (s < 0) {
        perror("Socket error");
        return 1;
    }

    // Set up the destination
    struct sockaddr_rc addr = {0};
    addr.rc_family = AF_BLUETOOTH;
    addr.rc_channel = (uint8_t) 1;
    str2ba(dest, &addr.rc_bdaddr);

    // Connect to the remote device
    int status = connect(s, (struct sockaddr *)&addr, sizeof(addr));
    if (status < 0) {
        perror("Connection error");
        close(s);
        return 1;
    }

    // Message to send
    const char* message = "Hello, Bluetooth!";

    // Send the message
    status = write(s, message, strlen(message));
    if (status < 0) {
        perror("Write error");
        close(s);
        return 1;
    }

    std::cout << "Message sent successfully!" << std::endl;

    // Close the socket
    close(s);

    return 0;
}

File: bluetooth_send.py
import socket

def connect():
    # Bluetooth device information
    server_mac_address = '24:0A:C4:29:67:CA'  # MAC address of the Bluetooth device

    # Create a Bluetooth socket
    sock = socket.socket(socket.AF_BLUETOOTH, socket.SOCK_STREAM, socket.BTPROTO_RFCOMM)

    # Connect to a Bluetooth device
    sock.connect((server_mac_address, 1))
    print("Connected to Bluetooth device")

    try:
        while True:
            # Send data
            message = input("Enter message: ")
            if message == "END":
                break
            sock.send(message.encode('utf-8'))
            print("Sent:", message)

            # Receive data from Bluetooth device
            data = sock.recv(1024)
            print("Received:", data.decode('utf-8'))

    finally:
        # Close the connection
        sock.close()

connect()
